db-api/index.js

```diff
@@ -0,0 +1,106 @@
+// required capabilities:
+// - CAP_NET_BIND_SERVICE (For listening on a port < 1024 so users can't steal its port number)
+// DO NOT run as root
+if(process.getuid()==0||process.geteuid()==0) {
+  throw new Error("Refusing to run as root. This should be running as an unprivileged user.");
+}
+import express from "express";
+import identHelper from "./ident-helper.js";
+import sql from "./database.js";
+const safe_regex = /^[A-Za-z0-9_-]+$/;
+
+function get_listen_port() {
+  if (
+    process.env.PORT &&
+    parseInt(process.env.PORT) > 0 &&
+    parseInt(process.env.PORT) < 65536
+  ) {
+    return parseInt(process.env.PORT);
+  }
+}
+const app = express();
+app.use(identHelper);
+app.use((req, res, next) => {
+  if (req.username == "root") {
+    return res.status(500).json({
+      message:
+        "Refusing to respond to request from root. Try accessing this API as an unprivileged user.",
+    });
+  }
+  if (req.username.includes("_")) {
```

Comment:
Why deny usernames with `_` in them but still include it in the safe regex?

Reply:
Because database names have to have `_` in them, but usernames should never so it doesn't look like a user owns someone else's database.

Reply:
Ah, I see :+1: